pkg/operator/ceph/file/filesystem_test.go

```diff
@@ -75,11 +75,26 @@ func TestCreateFilesystem(t *testing.T) {
 	mds.UpdateDeploymentAndWait, deploymentsUpdated = testopk8s.UpdateDeploymentAndWaitStub()
 
 	configDir, _ := ioutil.TempDir("", "")
+
 	// Output to check multiple filesystem creation
 	fses := `[{"name":"myfs","metadata_pool":"myfs-metadata","metadata_pool_id":1,"data_pool_ids":[2],"data_pools":["myfs-data0"]}]`
 
+	createdFs := `{"fs_name": "myfs", "metadata_pool": 2, "data_pools":[3]}`
+
+	firstGet := true
 	executor := &exectest.MockExecutor{
 		MockExecuteCommandWithOutputFile: func(command string, outFileArg string, args ...string) (string, error) {
+			if contains(args, "fs") && contains(args, "get") {
+				if firstGet {
+					firstGet = false
+					return "", errors.New("fs doesn't exist")
+				}
+				return createdFs, nil
+			} else if contains(args, "fs") && contains(args, "ls") {
+				return "[]", nil
+			} else if contains(args, "osd") && contains(args, "lspools") {
+				return "[]", nil
+			}
 			return "{\"key\":\"mysecurekey\"}", nil
```

Comment:
I will offer this suggestion, which you can implement optionally:

I realize that this isn't the convention that exists from the test as it was written before, but we should really be testing that the code is issuing the correct commands and that we see an error if Rook does something unexpected. I think it would be more correct to change this so that only a specific command returns the `...mysecurekey...` string and an unexpected command throws a panic. I don't know what or how many commands Rook normally executes for these tests, so this might be more annoying than just adding one extra `else if`.

Reply:
I needed to add a lot of comands in but I've structured it to the way you suggested.

I was surprised when I first came across this test that there wasn't stronger testing of how K8S specs map to Ceph commands. I certainly think there is value in testing this more strongly but I'd definitely do that in another change.